delivery_location/models.py

```diff
@@ -0,0 +1,53 @@
+from django.db import models
+from django.core.validators import MinLengthValidator
+from django.core.exceptions import ValidationError
+from supplier.models import Supplier
+
+
+class DeliveryLocation(models.Model):
+    user_name = models.ForeignKey(Supplier, on_delete=models.CASCADE)
+    location = models.CharField(max_length=32, validators=[MinLengthValidator(1)])
+    date = models.DateField()
+
+    def add_delivery_location(self):
+        self.save()
+        try:
+            DeliveryLocation.full_clean(self)
+        except ValidationError as e:
+            self.delete()
+            raise e
+        return self
+
+    def remove_delivery_location(self):
+        DeliveryLocation.objects.get(id=self.id).delete()
+
+    def update_date(self, newdate):
+        self.date = newdate
+        self.save()
+
+    def update_location(self, new_location):
+        if not new_location:
+            raise ValidationError("Please put a new location with length bigger than zero")
+        else:
+            self.location = new_location
+            self.save()
+
+    @staticmethod
+    def filter_by_supplier_and_location(supplier, specific_location):
+        return DeliveryLocation.objects.filter(user_name=supplier, location=specific_location)
+
+    @staticmethod
+    def filter_by_location(supplier):
+        return DeliveryLocation.objects.filter(user_name=supplier)
+
+    @staticmethod
+    def get_all_delivery_locations():
+        return DeliveryLocation.objects.all()
```

Comment:
omer meant that those two are irrelevant, no point warping those methods, the rest raises an error so i can understand.
technically if you create a class method ull need to test it, like omer said no point of doing so since they do the same as built in tested methods.

Reply:
These functions added so the logic of using the deliverylocation is known and in order to use that logic you just simply use the functions that is created and if there is functions that is needed in the future i will update the deliverylocation,Instead of writing the DeliveryLocation.objects.all() as a response of an http get request for example we use get_all_delivery_locations that is more understandable for example

Reply:
i still dont see the reason, get all delivery... is a static method, writing DeliveryLocation.get_all_delivery_location() is same as writing DeliveryLocation.objects.all(), both static and no extra feature was added.
if you believe theres a difference that i dont see, then ok🤷‍♂️

Reply:
Lets see you are a client that wants to check a code, which one is more understandable
object.all
or
get_all_delivery_locations

the same thing happend when we write code and trying to debug something if something goes wrong